examples/hif2a/fit_to_multiple_rbfes.py

```diff
@@ -120,9 +125,36 @@ def _blew_up(results: List[SimulationResult]) -> bool:
     # TODO: adjust this threshold a bit, move reliability calculations into fe/estimator.py or fe/model.py
     return np.isnan(du_dls).any() or (du_dls.std(1).max() > 1000)
 
-def loss_fxn(model, ff_params, mol_a, mol_b, core, label_ddG):
-    pred_ddG, stage_results = model.predict(ff_params, mol_a, mol_b, core)
-    return pseudo_huber_loss(pred_ddG - label_ddG), stage_results
+# def loss_fxn(model, ff_params, mol_a, mol_b, core, label_ddG):
+#     pred_ddG, stage_results = model.predict(ff_params, mol_a, mol_b, core)
+#     return pseudo_huber_loss(pred_ddG - label_ddG), stage_results
+
+def loss_fxn(ff_params, batch: List[Tuple[RelativeFreeEnergy, RBFEModel]]):
+    index = RBFETransformIndex()
+    index.build([edge[0] for edge in batch])
+    indices = []
+    all_results = []
+    preds = []
+    for rfe, model in batch:
+        indices.append(list(index.get_transform_indices(rfe)))
+        pred_ddG, stage_results = model.predict(ff_params, rfe.mol_a, rfe.mol_b, rfe.core)
+        all_results.extend(list(stage_results))
+        preds.append(pred_ddG)
+    # If the edges in the batch are within a cycle, correct the ddGs
+    indices = jnp.asarray(indices)
+    ind_l, ind_r = indices.T
+    try:
+        layer = construct_mle_layer(len(index), indices)
+        corrected_dg = layer(jnp.asarray(preds))
+        cycle_corrected_rbfes = corrected_dg[ind_r] - corrected_dg[ind_l]
+    except AssertionError:
+        # Provided non connected graph, ie its not a cycle
+        cycle_corrected_rbfes = jnp.asarray(preds)
+    labels = jnp.asarray([rfe.label for rfe, _ in batch])
+    loss = jnp.mean(jnp.abs(cycle_corrected_rbfes - labels))
+    # Aggregate using mean, then compute psuedo huber loss
+    loss = pseudo_huber_loss(loss)
```

Comment:
Could you comment a bit more on this step?

Would have expected to aggregate in the other order, perhaps as

```python
residuals = cycle_corrected_rbfes - labels
loss_terms = pseudo_huber_loss(residuals)
loss = mean(loss_terms)
```
or similar

Reply:
You are totally right, woops!

Reply:
Nice! Addressed in https://github.com/proteneer/timemachine/pull/449/commits/d89ff27d1e7a2a3a3507353c099a6164b8378867